docs/reference/ssl/swarm-with-traefik.md

```diff
@@ -0,0 +1,319 @@
+# SSL on Swarm with Traefik
+
+To completely secure your OpenFaaS installation, you need SSL.  On Swarm, you can do this easily with [Traefik][traefik] and [Let's Encrypt][letsencrypt].  Traefik is a reverse proxy that comes with SSL support via Let's Encrypt. In this tutorial we will show you how to deploy OpenFaaS with Traefik.
+
+
+## Create an A record
+
+If your domain is `.domain.com` then create an A record using your DNS administration panel such as `gateway.domain.com` or `openfaas.domain.com`. The required steps will vary depending on your domain provider and your cluster provider. For example; [on Google Cloud DNS](https://cloud.google.com/kubernetes-engine/docs/tutorials/configuring-domain-name-static-ip) or [with Route53 using AWS](https://kubernetes.io/docs/setup/custom-cloud/kops/#2-5-create-a-route53-domain-for-your-cluster).
+
+## Update the Compose configuration
+
+### Configure Traefik
+To use Traefik with OpenFaaS, you need to modify the OpenFaaS deployment manifest to include Traefik and configuring OpenFaaS to communicate through Traefik instead of directly exposing its services publicly.
+
+1. Clone OpenFaaS and then checkout the latest stable release:
+
+    ```bash
+    $ git clone https://github.com/openfaas/faas && cd faas
+    ```
+
+2. Add the `traefik` service to the  `docker-compose.yaml`.
```

Comment:
I can see why you've gone for this approach, but it's going to be difficult for a user to follow all the edits. My gist had a separate version of the file for ease of use, but this would mean some maintenance.

What do you think to defining a new file and referencing the network as external maybe? See the Kafka connector as an example of this